internal/pkg/sanitize/sanitize.go

```diff
@@ -36,33 +36,34 @@ func Sanitize(pkgPattern string, opts *Options) (*packages.OverlayJSON, error) {
 	}
 	log.Debugf("Working directory: %q\n", workDir)
 
-	for _, pkg := range pkgs {
+	packages.Visit(pkgs, nil, func(pkg *packages.Package) {
 		if !opts.shouldSanitize(pkg.PkgPath) {
-			continue
+			return
 		}
+		log.Debugf("Instrumenting package %q", pkg.PkgPath)
 		for i, sourceFile := range pkg.Syntax {
 			originalSourceFile := pkg.CompiledGoFiles[i]
 			transformer := hook.NewTransformer(sourceFile, pkg.Fset, originalSourceFile, pkg.TypesInfo, hook.NodeId)
 			if numHooks := transformer.TransformFile(); numHooks > 0 {
-				rel, err := fileutil.RelativePathInModule(pkg.Module.Dir, originalSourceFile)
+				baseFile := filepath.Base(originalSourceFile)
+				baseFileNoExtension := baseFile[:len(baseFile)-len(filepath.Ext(baseFile))]
+				instrumentedFile, err := os.CreateTemp(workDir, fmt.Sprintf("%s-*.go", baseFileNoExtension))
```

Comment:
IMO it'd be nice to have an additional indicator that this is an instrumented file, so we could generate file names like `%s-*.instr.go` or similar instead of just slapping a random ID on there. WDYT? Functionality-wise this doesn't change anything so feel free to resolve this if considered not necessary.

Reply:
We store the instrumented files in the working directory which is prefixed with `gofuzz-sanitize-` which IMHO gives hints that these are instrumented files. That said, adding an `instrumented` suffix to the file names could make this clear.

Reply:
Adjusted the pattern as proposed